url2b9f720f64_jfimbett_QBnB_tgz-pJ8-tests_TestJ8/src/algorithms/BnB.java

```diff
@@ -0,0 +1,412 @@
+package algorithms;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.ListIterator;
+
+import model.Model;
+import model.QBnBEnv;
+import model.QBnBVariable;
+import model.QBnBconstr;
+
+public class BnB {
+
+	/**
+	 * 1 for maximization
+	 * 0 for minimization
+	 */
+    private int type;
+	private Model best;
+	private Queue priorityQueue;
+	private double timeLoadingData;
+	private boolean print;
+	private double epsilon;
+	private boolean cuts;
+	private QBnBEnv e;
+	private int nodesExplored;
+	private boolean printSimplex;
+	private boolean debugmode;
+	
+	public BnB(Model initial, QBnBEnv e) {
+		this.e=e;
+		best=initial;
+		this.cuts=false;
+		
+	}
+
+	public void printInfoFromModel(Model m){
+		System.out.println("Solving Model: "+m.getId());
+		m.getSimplex().printProblem();
+	}
+	public void minimizes(){
+		nodesExplored=0;
+		double initSol=System.currentTimeMillis();
+		double bestSolution=Double.MAX_VALUE;
+		priorityQueue= new Queue<Model>();
+		priorityQueue.enqueue(best);
+		best.setId("");
+		int depth=0;
+		while(!priorityQueue.isEmpty()){
+			Model m= (Model)priorityQueue.dequeue();
+			//m.printProblemToConsole();
+			//System.out.println("Solving model "+m.getId());
+			m.solveLP(0);
+			double lowerbound=m.getSimplex().primalfo();
+			nodesExplored++;
+             if(m.getSimplex().getStatus()==3){
+				//System.out.println("Prunning not feasible branch");
+			}
+           else if(m.isInteger() && lowerbound<bestSolution){
+				best=m;
+				bestSolution=lowerbound;
+				if(print){
+					System.out.println("Best Integer Solution "+bestSolution);
+//					double dual=m.getSimplex().dualfo();
+//					double gap=(upperbound-dual)/upperbound;
+//					System.out.println("Gap "+gap*100+"%");
+					System.out.println("Integer Solution found, prunning rest of the branch");
+				}
+				
+			}
+			
+			else if(lowerbound<bestSolution){
+			
+				Model[] aa= integerBranch(m);
+				Model a=aa[0];
+				
+				a.setId(m.getId()+"l");
+				priorityQueue.enqueue(a);	
+				
+				Model b=aa[1];
+				b.setId(m.getId()+"r");
+				priorityQueue.enqueue(b);
+				
+				
+				
+			}
+			else{
+				//System.out.println("Prunning for bound");
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 72:89, 140:151.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.